heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -334,9 +339,14 @@ private void removeInstancesFromContainers(PackingPlanBuilder packingPlanBuilder
    * Remove an instance of a particular component from the containers
    */
   private void removeRRInstance(PackingPlanBuilder packingPlanBuilder,
-                                String component) throws RuntimeException {
-    ContainerIdScorer scorer = new ContainerIdScorer(this.containerId, this.numContainers);
-    this.containerId = nextContainerId(packingPlanBuilder.removeInstance(scorer, component));
+                                String componentName) throws RuntimeException {
+    List<Scorer<Container>> scorers = new ArrayList<>();
+    scorers.add(new HomogeneityScorer(componentName, true));  // all-same-component containers first
+    scorers.add(new InstanceCountScorer());                   // then fewest instances
+    scorers.add(new HomogeneityScorer(componentName, false)); // then least homogeneous
```

Comment:
Good catch. That's a bug in my comment, we are doing most homogeneous first. Updated.